pkg/transforms/genericresource.go

```diff
@@ -33,33 +33,57 @@ func GenericResourceBuilder(r *unstructured.Unstructured) *GenericResource {
 	group := r.GroupVersionKind().Group
 	kind := r.GetKind()
 	transformConfig, found := getTransformConfig(group, kind)
-	if found {
-		for _, prop := range transformConfig.properties {
-			jp := jsonpath.New(prop.name)
-			parseErr := jp.Parse(prop.jsonpath)
-			if parseErr != nil {
-				klog.Errorf("Error parsing jsonpath [%s] for [%s.%s] prop: [%s]. Reason: %v",
-					prop.jsonpath, kind, group, prop.name, parseErr)
-				continue
-			}
-			result, err := jp.FindResults(r.Object)
-			if err != nil {
-				// This error isn't always indicative of a problem, for example, when the object is created, it
-				// won't have a status yet, so the jsonpath returns an error until controller adds the status.
-				klog.V(1).Infof("Unable to extract prop [%s] from [%s.%s] Name: [%s]. Reason: %v",
-					prop.name, kind, group, r.GetName(), err)
-				continue
-			}
-			if len(result) > 0 && len(result[0]) > 0 {
-				n.Properties[prop.name] = fmt.Sprintf("%s", result[0][0])
-			} else {
-				klog.Errorf("Unexpected error extracting [%s] from [%s.%s] Name: [%s]. Result object is empty.",
-					prop.name, kind, group, r.GetName())
-				continue
-			}
+
+	// Currently, only pull in the additionalPrinterColumns listed in the CRD if it's a Gatekeeper
+	// constraint. Simply removing the `group == "constraints.gatekeeper.sh"` below will enable it
+	// for all other API groups.
+	if !found && group == "constraints.gatekeeper.sh" {
+		transformConfig = ResourceConfig{properties: additionalColumns}
+	}
+
+	for _, prop := range transformConfig.properties {
+		// Skip properties that are already set. This could happen if additionalPrinterColumns
```

Comment:
So, we have a few properties that are a list of stings. It would cause problems if the same property name is used by a different resource as a single string.
It is a short list, of the top of my head some of these properties are 
- container (Pod)
- images (Pod)
- role (Node)
- there's a few more, I'll need to search all transforms for any property defined as string[].

To avoid problems we would need to either:
- ignore properties with this name OR
- wrap the value in a string[]

Reply:
I think I got them all in the latest push, but let me know if I missed any!